### PR TITLE
Update tap/hold actions to objects

### PR DESCRIPTION
**Pull request by iantrich**
https://github.com/home-assistant/home-assistant-polymer/pull/2182
https://github.com/home-assistant/home-assistant-polymer/pull/2186
https://github.com/home-assistant/home-assistant-polymer/pull/2187

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_lovelace/entity-button.markdown [hunk 1]
```diff
@@ -47,23 +47,54 @@ icon:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/entity-button.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/entity-button.markdown [hunk 2]
```diff
@@ -47,23 +47,54 @@ icon:
...
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
-  default: "entity_id: entity_id"
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/entity-button.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/glance.markdown [hunk 3]
```diff
@@ -69,23 +69,54 @@ icon:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/glance.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/glance.markdown [hunk 4]
```diff
@@ -69,23 +69,54 @@ icon:
...
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
-  default: "entity_id: entity_id"
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/glance.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 5]
```diff
@@ -92,26 +71,54 @@ entity:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 6]
```diff
@@ -92,26 +71,54 @@ entity:
...
-  required: false
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 7]
```diff
@@ -140,26 +147,54 @@ suffix:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 8]
```diff
@@ -140,26 +147,54 @@ suffix:
...
-  required: false
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture.markdown [hunk 9]
```diff
@@ -26,18 +26,56 @@ image:
...
   required: false
-  description: The service data.
+  description: Action to take on tap-and-hold
   type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture.markdown [hunk 10]
```diff
@@ -26,18 +26,56 @@ image:
...
-  type: string
-service_data:
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-glance.markdown [hunk 11]
```diff
@@ -59,6 +55,56 @@ entity:
...
+hold_action:
+  required: false
+  description: Action to take on tap-and-hold
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-glance.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-glance.markdown [hunk 12]
```diff
@@ -59,6 +55,56 @@ entity:
...
+tap_action:
+  required: false
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-glance.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-entity.markdown [hunk 13]
```diff
@@ -58,27 +58,54 @@ show_state:
...
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
-  default: "`entity_id: entity_id`"
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-entity.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-entity.markdown [hunk 14]
```diff
@@ -58,27 +58,54 @@ show_state:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-entity.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 15]
```diff
@@ -254,26 +317,54 @@ entity:
...
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 16]
```diff
@@ -254,26 +317,54 @@ entity:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 17]
```diff
@@ -214,26 +249,54 @@ entity:
...
-  required: false
-  description: The service data to use.
+  description: Action to take on tap-and-hold
   type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

File: source/_lovelace/picture-elements.markdown [hunk 18]
```diff
@@ -214,26 +249,54 @@ entity:
...
-  type: string
-  default: more-info
+  description: Action to take on tap
+  type: object
+  keys:
+    action:
+      required: true
+      description: Action to perform (`more-info`, `toggle`, `call-service`, `navigate`, `none`)
+      type: string
+      default: `none`
```
Comment (klaasnicolaas) [inline source/_lovelace/picture-elements.markdown]:
```suggestion
      default: "`none`"
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
I left some suggestions to fix the build error, they are all the same 😋

File: source/_lovelace/picture-elements.markdown (hunk 18 above)
Comment (iantrich) [reply source/_lovelace/picture-elements.markdown]: @klaasnicolaas why do you think they should be quotes surrounding it?

**Comment by balloob** [comment]
oops it was broken.

**Pull request merged**